index.js

```diff
@@ -140,39 +160,57 @@ const redisStore = (...args) => {
           cb = (err, result) => (err ? reject(err) : resolve(result));
         }
 
-        redisCache.del.apply(redisCache, [...args, handleResponse(cb)]);
+        const adjustedKeys = prepareKeys(cacheName, args);
+        redisCache.del.apply(redisCache, [...adjustedKeys, handleResponse(cb)]);
       })
     ),
     reset: cb => (
-      new Promise((resolve, reject) => {
-        if (!cb) {
-          cb = (err, result) => (err ? reject(err) : resolve(result));
-        }
-  
-        redisCache.flushdb(handleResponse(cb));
-      })
+        new Promise(async (resolve, reject) => {
+            if (!cb) {
+                cb = (err, result) => (err ? reject(err) : resolve(result));
+            }
+
+            let allKeys
+            try {
+                allKeys = await getKeys(redisCache, cacheName, '*', undefined);
+            } catch(error) {
+                return handleResponse(cb, {keys: []})(error);
+            }
+            if(allKeys && allKeys.length > 0) {
+                redisCache.del.apply(redisCache, [...prepareKeys(cacheName, allKeys), handleResponse(cb, {keys: allKeys})]);
+            } else {
+                return handleResponse(cb, {keys: []})(null);
+            }
+        })
     ),
     keys: (pattern = '*', cb) => (
-      new Promise((resolve, reject) => {
-        if (typeof pattern === 'function') {
-          cb = pattern;
-          pattern = '*';
-        }
-
-        if (!cb) {
-          cb = (err, result) => (err ? reject(err) : resolve(result));
-        }
-
-        redisCache.keys(pattern, handleResponse(cb));
-      })
+        new Promise(async (resolve, reject) => {
+            if (typeof pattern === 'function') {
+                cb = pattern;
+                pattern = '*';
+            }
+
+            if (!cb) {
+                cb = (err, result) => (err ? reject(err) : resolve(result));
+            }
+
+            try {
+                const keys = await getKeys(redisCache, cacheName, pattern, undefined);
+
+                return handleResponse(cb, {keys})(null, keys);
```

Comment:
you probably don't want to `return` here anything, you are inside the body of a promise